src/ecalc/libraries/libecalc/common/libecalc/input/yaml_types/components/category.py

```diff
@@ -0,0 +1,34 @@
+from typing import Any, Literal
+
+from pydantic import Field
+
+CategoryType = Literal[
```

Comment:
Instead of hard coding: Maybe import the user defined categories from dto.base?
Then user_defined_categories = Union[ConsumerUserDefinedCategoryType, InstallationUserDefinedCategoryType, FuelTypeUserDefinedCategoryType].